File: ClearFiveStages-HTML5/GameDataManager.js
var LANGUAGE_CHINESE = 1;
var LANGUAGE_JAPANESE = 2;
var LANGUAGE_ENGLISH = 3;
var GAMELEVEL_EASY = 1;
var GAMELEVEL_NORMAL = 2;
var GAMELEVEL_HARD = 3;

var GameDataManager = function () {};
GameDataManager.languageType = LANGUAGE_ENGLISH;
GameDataManager.gameLevel = GAMELEVEL_EASY;
GameDataManager.tableBgIndex = 2;
GameDataManager.isMusicOn = true;
GameDataManager.isSoundEffectOn = true;
GameDataManager.playerName = "Raccoon";
GameDataManager.hintCount = 0;

GameDataManager.loadSetting= function () {
    GameDataManager.languageType = LANGUAGE_CHINESE;
};

GameDataManager.updatePlayerName= function (name) {
    if (name == "") return;
    if (name == GameDataManager.playerName) return;
    GameDataManager.playerName = name;
};

GameDataManager.updateGameLevel = function (level) {
    if (GameDataManager.gameLevel == level) return;
    GameDataManager.gameLevel = level;
};

GameDataManager.resetHintCount = function () {
	if (GameDataManager.gameLevel == GAMELEVEL_EASY) GameDataManager.hintCount = 10;
	if (GameDataManager.gameLevel == GAMELEVEL_NORMAL) GameDataManager.hintCount = 5;
	if (GameDataManager.gameLevel == GAMELEVEL_HARD) GameDataManager.hintCount = 0; 
};

File: ClearFiveStages-iOS/ClearFiveStages/Manager/GameDataManager.m
#import "GameDataManager.h"
#import "SettingsManager.h"

@implementation GameDataManager

@synthesize languageType;
@synthesize gameLevel;
@synthesize tableBgIndex;
@synthesize isMusicOn;
@synthesize isSoundEffectOn;
@synthesize playerName;
@synthesize hintCount;

static GameDataManager *_sharedManager = nil;

//Init Method
+(GameDataManager *) sharedManager {
	if (!_sharedManager) {
		if( [ [GameDataManager class] isEqual:[self class]] )
			_sharedManager = [[GameDataManager alloc] init];
		else
			_sharedManager = [[self alloc] init];
	}
	return _sharedManager;
}

-(id) init {  
	if( (self=[super init]) ) {
		gameLevel = GAMELEVEL_EASY;
		hintCount = 10;
		isMusicOn = YES;
		isSoundEffectOn = YES;
        NSUserDefaults* defs = [NSUserDefaults standardUserDefaults];
        NSArray* languages = [defs objectForKey:@"AppleLanguages"];
        NSString* preferredLang = [languages objectAtIndex:0];
        self.languageType = LANGUAGE_ENGLISH;
        self.playerName = [defs objectForKey:@"PlayerName"] ? [defs objectForKey:@"PlayerName"] : @"PLAYER"; 
        if ([preferredLang isEqualToString:@"ja"]) self.languageType = LANGUAGE_JAPANESE;
        if ([preferredLang hasPrefix:@"zh-"]) self.languageType = LANGUAGE_CHINESE;
	}
	return self;
}

-(void) updateGameLevel:(GameLevel)level {
    if (gameLevel == level) return;
    gameLevel = level;
}

-(void) resetHintCount {
	if (gameLevel == GAMELEVEL_EASY) hintCount = 10;
	if (gameLevel == GAMELEVEL_NORMAL) hintCount = 5;
	if (gameLevel == GAMELEVEL_HARD) hintCount = 0; 
}

-(void) updatePlayerName:(NSString *)name {
    if ([name isEqualToString:@""]) return;
    if ([name isEqualToString:self.playerName]) return;
    self.playerName = name;
    [[NSUserDefaults standardUserDefaults] setObject:self.playerName forKey:@"PlayerName"]; 
}

-(void) dealloc {
	 [super dealloc];
}
@end
